src/python/pants/util/dirutil.py

```diff
@@ -125,32 +127,66 @@ def safe_file_dump(filename, payload, binary_mode=None, mode=None):
     else:
       mode = 'wb'
 
-  with safe_open(filename, mode=mode) as f:
-    f.write(payload)
+  safe_file_write(filename, payload=payload, mode=mode)
 
 
-def maybe_read_file(filename, binary_mode=True):
+def safe_file_write(filename, payload=None, mode=None):
```

Comment:
Why not put the defaults in signature?
```suggestion
def safe_file_write(filename, payload='', mode='w'):
```

One benefit is it allows editors/IDEs to preview what the function defaults are.

Reply:
I was thinking that `=None` is nice if arguments are to be forwarded from other python methods, but that's more relevant with objects methods, not free functions. Will fix this.

Reply:
Done!